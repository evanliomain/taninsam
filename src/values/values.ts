/**
 * @module object=>array
 */
import { chain } from '../chain';
import { keys } from '../keys';
import { map } from '../map';

/**
 * This method extract own enumerable values of an object into an array
 * See the [native Object.values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/values) for more informations.
 * @return the function to apply on the object to extract its values
 * @example
 * ```
 * values()({ a: 1, b: 2, c: 3 }) // [1, 2, 3]
 * ```
 * @example Using the chain
 * ```
 * chain({ a: 1, b: 2, c: 3 })
 *   .chain(values())
 *   .value() // [1, 2, 3]
 * ```
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function values(): (ob: object) => ReadonlyArray<any> {
  return (ob: object) =>
    chain(ob)
      .chain(keys())
      .chain(map(key => ob[key]))
      .value();
}
