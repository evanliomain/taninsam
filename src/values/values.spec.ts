import { values } from './values';

describe('values function', () => {
  const input = { a: 1, b: 2, c: 3 };

  test('is a pure function', () => {
    values()(input);
    expect(input).toMatchSnapshot();
  });

  test('{ a: 1, b: 2, c: 3 } |> values === [1, 2, 3]', () => {
    expect(values()(input)).toMatchSnapshot();
  });

  test('{} |> values === []', () => {
    expect(values()({})).toMatchSnapshot();
  });
});
