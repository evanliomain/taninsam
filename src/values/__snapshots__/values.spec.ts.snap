// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`values function > { a: 1, b: 2, c: 3 } |> values === [1, 2, 3] 1`] = `
[
  1,
  2,
  3,
]
`;

exports[`values function > {} |> values === [] 1`] = `[]`;

exports[`values function > is a pure function 1`] = `
{
  "a": 1,
  "b": 2,
  "c": 3,
}
`;
